### PR TITLE
#869 introduce cached tojos

### PR DESCRIPTION
**Pull request by mximp**
#869

### CODE REVIEW

**Comment by mximp** [comment]
@yegor256 please review and merge

File: src/main/java/com/yegor256/tojos/CachedTojos.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    @Override
+    public Tojo add(final String name) {
+        cache.clear();
+        return wrapped.add(name);
+    }
+
+    @Override
+    public List<Tojo> select(final Predicate<Tojo> filter) {
+        if (this.cache.isEmpty()) {
```
Comment (yegor256) [inline src/main/java/com/yegor256/tojos/CachedTojos.java:63]: @mximp this class is not thread-safe with this design. We should either say it in the Javadoc, or make it thread-safe. I would go with the second option.

**Comment by yegor256** [comment]
@mximp looks good, just one comment above

**Comment by mximp** [comment]
@yegor256 done

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by mximp** [comment]
@yegor256 please try to merge again

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**